src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs

```diff
@@ -276,6 +277,7 @@ private async Task<bool> WriteResultsAsync(TContext rootContext)
 
                             RuntimeErrors |= context.RuntimeErrors;
 
+                            context.Dispose();
```

Comment:
>context.Dispose(); [](start = 28, length = 18)

put this in a finally block! i.e., test for null and dispose if not so. the reason is that your code currently doesn't dispose in the exceptional case. one thing, though! if you do this, remove the context = default in the exception block, as it might prevent disposal! (finally will happen after the catch). alternately, just add the dispose to the catch block. :)

Reply:
I saw this BUT, since the throwExitApplication will generate a new exception and stop the application I think that is not needed.
What is ur opinion?

---
In reply to: [584349903](https://github.com/microsoft/sarif-sdk/pull/2302#discussion_r584349903) [](ancestors = 584349903)